### PR TITLE
(maint) Merge 2.3.1 into master (fact paths, etc.)

### PR DESCRIPTION
**Pull request by rbrw**
```
    (maint) Merge 2.3.1 into master (fact paths, etc.)
    
    * stable:
      (PDB-1328) Add contributors to and edit release notes
      (PDB-1328) Add preliminary 2.3.1 release notes
      (PDB-1031) Verify gc behavior in update race test
      (PDB-1031) Reproduce the fact paths constraint issue
      (PDB-1031) Speed migration with unique facts table
      (PDB-1031) Accommodate move of fact paths to facts
      (PDB-1031) Fix query engine, v3 endpoints for fact_path move
      (docs) Replace Checkin with Check In
      (PDB-1342) Fix trusted facts fallback
      (docs) Add .html suffixes to 2.3.0 release notes
      (PDB-1307) Update to tk-jetty 1.3.0
      (maint) applied-migrations: don't ignore exceptions

 documentation/release_notes.markdown                   |  87 +++++-
 documentation/repl.markdown                            |   3 +-
 documentation/scaling_recommendations.markdown         |   2 +-
 project.clj                                            |   2 +-
 puppet/lib/puppet/indirector/facts/puppetdb.rb         |   2 +-
 puppet/spec/unit/indirector/facts/puppetdb_spec.rb     |  25 +-
 src/puppetlabs/puppetdb/cli/services.clj               |   5 +-
 src/puppetlabs/puppetdb/facts.clj                      |  90 +++---
 src/puppetlabs/puppetdb/query.clj                      |   2 +-
 src/puppetlabs/puppetdb/query_eng/engine.clj           |  22 +-
 src/puppetlabs/puppetdb/scf/migrate.clj                | 129 +++++++--
 src/puppetlabs/puppetdb/scf/migration_legacy.clj       | 293 +++++++++++++++++++
 src/puppetlabs/puppetdb/scf/storage.clj                | 576 ++++++++++++++++++-------------------
 test/puppetlabs/puppetdb/command_test.clj              | 155 +++++++++-
 test/puppetlabs/puppetdb/facts_test.clj                |  72 -----
 test/puppetlabs/puppetdb/scf/migrate_test.clj          | 100 ++++++-
 test/puppetlabs/puppetdb/scf/migration_legacy_test.clj |  76 +++++
 test/puppetlabs/puppetdb/scf/storage_test.clj          | 157 ++++++++--
 18 files changed, 1303 insertions(+), 495 deletions(-)
```

### CODE REVIEW

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -302,6 +302,73 @@ derive data from a PuppetDB export tarball.
 
 TODO
```
Comment (wkalt) [inline documentation/release_notes.markdown]: @rlb does this need to go?

Comment (rbrw) [inline documentation/release_notes.markdown]: @wkalt perhaps not, that's a placeholder (already) in master, though I think we may eventually want to move it back up near the top where we normally put the contributors?

Comment (kbarber) [inline documentation/release_notes.markdown]: leave it, pre-existing & out of scope.

**Comment by wkalt** [comment]
this looks good to me on first glance

**Pull request merged**